src/components/button/button.ts

```diff
@@ -129,10 +118,8 @@ export class SbbButtonElement extends LitElement implements LinkButtonProperties
     this._handlerRepository.disconnect();
   }
 
-  private _onLabelSlotChange(event: Event): void {
-    this._hasText = (event.target as HTMLSlotElement)
-      .assignedNodes()
-      .some((n) => !!n.textContent?.trim());
+  private _onLabelSlotChange(): void {
+    this._hasText = this._namedSlots.slots.has('unnamed');
```

Comment:
Maybe the field this._hasText is not needed anymore and so also this method at all? @TomMenga

Reply:
The answer is the same for both comments
We need the `@state _hasText` to trigger a new render of the component

We could argue that the `data-icon-only` could be replaced by a CSS rule, but I think that would make it less readable